internal/artifacts/module_arch.go

```diff
@@ -464,18 +464,20 @@ func archiveModuleToResultDir(buildResult string, requestedResultFileName string
 }
 
 // getIgnores - get files and/or subfolders to exclude from the package.
-func getIgnores(moduleLoc dir.ITargetModule, module *mta.Module) []string {
+func getIgnores(moduleLoc dir.IModule, module *mta.Module) []string {
 	var ignoreList []string
 	// ignore defined in build params is declared
 	if module.BuildParams != nil && module.BuildParams[ignore] != nil {
 		ignoreList = convert(module.BuildParams[ignore].([]interface{}))
 	}
-	// we add temporary folder to the list of ignores, because
-	// for the case when source == target and module path == project path,
-	// temporary folder can become part of the packaged module content
-	tmpFolder := filepath.Dir(moduleLoc.GetTargetModuleDir(module.Name))
-	tmpFolderName := filepath.Base(tmpFolder)
-	ignoreList = append(ignoreList, tmpFolderName)
+	// we add target folder to the list of ignores to avoid it's packaging
+	// it can be the case only when target folder is subfolder of the project path
+	targetFolder := moduleLoc.GetTargetTmpDir()
+	sourceFolder := moduleLoc.GetSource()
+	if strings.HasPrefix(targetFolder, sourceFolder) {
```

Comment:
I think it would be better to use `filepath.Rel` instead of checking a string prefix, which could cause bugs in some edge cases (and in more cases if the paths are not absolute or not normalized). To check if it's inside the folder you have to make sure `filepaths.Rel` result doesn't start with `../` (depending on the separator of course) or equals `..`.

Edge case where using string prefix will cause a bug:
`sourceFolder = a`
`targetFolder = abc`
The result will be `bc` which is incorrect. `filepath.Rel` should return `../abc` in this case.